routers/record.go

```diff
@@ -78,5 +75,15 @@ func AfterRecordMessage(ctx *context.Context) {
 	if userId != "" {
 		record.Organization, record.User = util.GetOwnerAndNameFromId(userId)
 	}
-	util.SafeGoroutine(func() { object.AddRecord(record) })
+
+	recordSignup := ctx.Input.Params()["recordSignup"]
+	if recordSignup == "true" {
+		record2 := *record
+		record2.Action = "signup"
```

Comment:
What if record2 is nil?

Reply:
I use * to copy record because record is a pointer ,which means that record is type `*casvisorsdk.Record` , so record2 will never be nil